feat: Misc Dev Portal updates from previous feedback

cloudjumpercat opened the pull request:
### Description

- [Enable or Disable Application Registration for an API Product Version - Kong Konnect | Kong Docs](https://docs.konghq.com/konnect/dev-portal/applications/enable-app-reg)  “Not really related to your changes, but the section above this ( Support for any control plane ) seems a bit out of place.” 
- The original comment has been lost to the ether, but Angel mentioned moving the section about developer app analytics that was on [this page](https://docs.konghq.com/konnect/dev-portal/access-and-approval/manage-app-reg-requests/) to the Dev Portal overview (which I did in the multi-portal PR) and then removing this page (which I didn’t do).
  - I think the ideal solution is to compound the remaining info on that page into just a short blurb about how to manage app requests and add it to this page [Manage Application Registrations ](https://docs.konghq.com/konnect/dev-portal/access-and-approval/manage-app-connections/)  and then remove [Manage Application Registration Requests](https://docs.konghq.com/konnect/dev-portal/access-and-approval/manage-app-reg-requests/)
  - Also, I noticed [Auto Approve Developer and Application Registrations ](https://docs.konghq.com/konnect/dev-portal/access-and-approval/auto-approve-devs-apps/)  is slightly similar. Can we revise all three docs into just one page about managing dev app registrations (auto approve, initial approval, revoke an existing approved app)? 
- From Amy: “what do you think about using "authentication strategy" rather than "auth strategy", except when referring to UI use of that term?” :check_mark: (where applicable, I renamed the first instance to “auth strategy” since I think customers would be familiar with the shortened version)
- Revisit Dev Portal Overview diagram (tried to find in old commits, but this is also lost to the ether) (made a new [Dev Portal config prep page](https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/configuration-prep/))
 
<!-- Include any supporting resources, e.g. link to a Jira ticket, GH issue, FTI, Slack, Aha, etc. -->
DOCU-3951
### Testing instructions

Preview links:
- NEW: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/configuration-prep/
- REMOVED: https://docs.konghq.com/konnect/dev-portal/access-and-approval/auto-approve-devs-apps/ (this info is already in https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/create-dev-portal/)
- REVISED: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/access-and-approval/manage-app-connections/
- REMOVED: https://docs.konghq.com/konnect/dev-portal/access-and-approval/manage-app-reg-requests/ (added content to the doc listed above this one)
- REVISED: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/applications/application-overview/
- REVISED: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/applications/enable-app-reg/
- REVISED: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/

### Checklist 

- [x] Review label added <!-- (see below) -->
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

<!-- For example, if this change is for an upcoming 3.6 release, enclose your content in `{% if_version gte:3.6.x %} <content> {% endif_version %}` tags. 

Use any of the following keys:
* `gte:<version>` - greater than or equal to a specific version
* `lte:<version>` - less than or equal to a specific version
* `eq:<version>` - exactly equal to a specific version

You can do the same for older versions. -->

<!-- !!! Only Kong employees can add labels due to a GitHub limitation. If you're an OSS contributor, thank you! The maintainers will label this PR for you !!! -->

<!-- When raising a pull request, indicate what type of review you need with one of the following labels:

    review:copyedit: Request for writer review.
    review:general: Review for general accuracy and presentation. Does the doc work? Does it output correctly?
    review:tech: Request for technical review for a docs platform change.
    review:sme: Request for review from an SME (engineer, PM, etc).

At least one of these labels must be applied to a PR or the build will fail.
-->

cloudjumpercat commented on app/konnect/dev-portal/configuration-prep.md:
@lmilan Here's the Dev Portal diagram doc we were talking about in ticket review. If you have the time, I'd love a very general/high level review of [this page](https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/configuration-prep/):
- Does the page seem helpful? Were you able to understand the information (some of the text is rough because I was just trying to get the idea out)
- Would this info be better on a different page rather than a separate page?
- Do the diagrams make sense and do they cover most of the configuration situations?
- Are the diagrams needed or could the info be presented in another way better?

cc @Guaris and @lena-larionova if you wanted to review this as well

lena-larionova replied:
For "Who will be using your Dev Portal or Dev Portals?", why do we need three diagrams? It seems like the third diagram, the mix, covers all three scenarios in one diagram. I think you could use just the third one with a couple of minor tweaks.

lena-larionova replied:
Looking at this from another step back, I think you can condense the whole page to one or two diagrams. The section "What will my Dev Portal be used for?" looks like it contains the diagram for "Who will be using your Dev Portal or Dev Portals?". So you could lead the user down the page this way:

1. What is the dev portal for?
2. Who is using it?
3. What do I do with their apps, if any?

cloudjumpercat commented on app/_redirects:
```diff
@@ -93,6 +93,8 @@
 /konnect/dev-portal/dev-reg/    /konnect/dev-portal/register-and-create-app/
 /konnect/dev-portal/applications/dev-apps/    /konnect/dev-portal/register-and-create-app/
 /konnect/dev-portal/applications/dev-gen-creds/    /konnect/dev-portal/register-and-create-app/
+/dev-portal/access-and-approval/manage-app-reg-requests/    /dev-portal/access-and-approval/manage-app-connections/
+/dev-portal/access-and-approval/auto-approve-devs-apps/ /dev-portal/create-dev-portal/
```
```suggestion
/konnect/dev-portal/access-and-approval/manage-app-reg-requests/    /konnect/dev-portal/access-and-approval/manage-app-connections/
/konnect/dev-portal/access-and-approval/auto-approve-devs-apps/ /konnect/dev-portal/create-dev-portal/
```

cloudjumpercat commented on app/konnect/dev-portal/access-and-approval/manage-app-connections.md:
```diff
@@ -3,67 +3,29 @@ title: Manage Application Registrations
+
+When a developer [registers an application with an API product](/konnect/dev-portal/register-and-create-app),
+the requests must be approved by an admin if
+[auto approve](/dev-portal/create-dev-portal/) isn't enabled. When
```
```suggestion
[auto approve](/konnect/dev-portal/create-dev-portal/) isn't enabled. When
```

cloudjumpercat commented on app/konnect/dev-portal/applications/enable-app-reg.md:
```diff
@@ -69,7 +61,7 @@ For non-`default` control planes, app registration is supported using the `konne
 1. Enable **Accept application registrations** for each Dev Portal to allow developers to register their applications to consume this API.
 
-5. Optional: Click the [**Auto Approve**](/konnect/dev-portal/access-and-approval/auto-approve-devs-apps/) checkbox if you want to automatically approve all application registration requests.
+5. Optional: Click the [**Auto Approve**](/dev-portal/create-dev-portal/) checkbox if you want to automatically approve all application registration requests.
```
```suggestion
5. Optional: Click the [**Auto Approve**](/konnect/dev-portal/create-dev-portal/) checkbox if you want to automatically approve all application registration requests.
```

cloudjumpercat commented on app/konnect/dev-portal/applications/enable-app-reg.md:
```diff
@@ -89,7 +81,7 @@ If you already have an OIDC Auth Strategy created in the **Application Auth**:
 1. Enable **Accept application registrations** for each Dev Portal to allow developers to register their applications to consume this API.
 
-5. Optional: Click the [**Auto Approve**](/konnect/dev-portal/access-and-approval/auto-approve-devs-apps/) checkbox if you want to automatically approve all application registration requests.
+5. Optional: Click the [**Auto Approve**](/dev-portal/create-dev-portal/) checkbox if you want to automatically approve all application registration requests.
```
```suggestion
5. Optional: Click the [**Auto Approve**](/konnect/dev-portal/create-dev-portal/) checkbox if you want to automatically approve all application registration requests.
```

cloudjumpercat commented on app/konnect/dev-portal/applications/enable-app-reg.md:
```diff
@@ -120,7 +112,7 @@ If you do _not_ already have an OIDC Auth Strategy created, we will first create
 1. Enable **Accept application registrations** for each Dev Portal to allow developers to register their applications to consume this API.
 
-5. Optional: Click the [**Auto Approve**](/konnect/dev-portal/access-and-approval/auto-approve-devs-apps/) checkbox if you want to automatically approve all application registration requests.
+5. Optional: Click the [**Auto Approve**](/dev-portal/create-dev-portal/) checkbox if you want to automatically approve all application registration requests.
```
```suggestion
5. Optional: Click the [**Auto Approve**](/konnect/dev-portal/create-dev-portal/) checkbox if you want to automatically approve all application registration requests.
```

cloudjumpercat commented:
@Mierenga @c3pko Hi! I made a bunch of misc Dev Portal doc revisions, most came out of feedback that I didn't have enough time to act on for multi-portal (see the PR description for more info). If you have the time, could you do a PR review for technical accuracy? Specifically, I'd love some eyes on the new [Dev Portal config prep page](https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/configuration-prep/) (is it helpful? did I cover the common config scenarios? Is anything technically wrong?). Here's a list of all the changes:

- NEW: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/configuration-prep/
- REMOVED: https://docs.konghq.com/konnect/dev-portal/access-and-approval/auto-approve-devs-apps/ (this info is already in https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/create-dev-portal/)
- REVISED: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/access-and-approval/manage-app-connections/
- REMOVED: https://docs.konghq.com/konnect/dev-portal/access-and-approval/manage-app-reg-requests/ (added content to the doc listed above this one)
- REVISED: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/applications/application-overview/
- REVISED: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/applications/enable-app-reg/
- REVISED: https://deploy-preview-7638--kongdocs.netlify.app/konnect/dev-portal/

Mierenga commented on app/konnect/dev-portal/applications/application-overview.md:
```diff
@@ -20,7 +20,37 @@ You can perform the following actions from an application's details page:
+### Support for any control plane
+
+App registration is fully supported in the `default` control plane when using the application `consumers` and the `acl` plugin. The `default` control plane is the one that is first created in each geo when you create an organization.
+For non-`default` control planes, app registration is supported using the `konnect-application-auth` plugin available as of {{site.base_gateway}} 3.0. <!-- We mention using the different plugins, what do we need to do in those plugins to support app registration? Why do default CP and non-default CPs use different plugins and why those particular plugins? -->
```
This is indeed a complicated part or the product to explain. Here is some context and explanation:

Initially app registration in Konnect was implemented using a combination of `acl`, `key-auth`, and `openid-connect` plugins which are included in every version of Kong Gateway that is supported by Konnect. Using these plugins gives the benefits that they are compatible with Gateway `consumers` which can be mapped to portal applications, providing further plugin functionality like applying `rate-limiting` plugins based on the application `consumer`, etc.

This feature was developed when Konnect had only a single Control Plane per organization.

However, at the time that Konnect rolled out support for multiple Control Planes, the application registration feature was re-imagined so that it could scale to support APIs that might live in different control planes. As part of this, the plugins, keys, and consumers previously used were dropped from the design and replaced with a new plugin `konnect-application-auth`, credential management system, and access control system.

Migrating existing customers who were using the "legacy" (consumer-based) app registration design was not fully possible, however, because:
- Some customers were relying on the use of application `consumers` which are no longer supported
- The new design (and `konnect-application-auth` plugin) is only available on Kong Gateway versions 3.0+, and some customers were unable to upgrade from a 2.x version

This left the system in the following state:
- the (first) control plane in all existing orgs became known as the `default` control plane which would continue to support certain legacy behaviors (including consumer-based app reg) that would not be supported in newly created control planes
- new orgs would still have their first control plane be considered `default` and use the legacy consumer-based app registration
- additional control planes created in any org would be considered "non-default"/non-legacy and use the new design centered around the `konnect-application-auth` plugin. These control planes are not compatible with Kong Gateway versions < 3.0

This has been the case up until recently, where an update was made so that new orgs will no longer receive a control plane that is considered "default" or legacy, and there is no longer support for using Kong Gateway < 3.0 with app registration in new orgs. The exception is "legacy mode" for app registration can still be turned on for any control plane if requested via support.

I hope that helps to clarify some of the context and I would be happy to discuss this more if needed.

cloudjumpercat replied:
@Mierenga Thanks for the info! This is really helpful. Just to make sure I understand, the content I currently had would only apply to legacy users (whether they got Konnect before 3.0 or because they requested legacy through support), this wouldn't apply to new orgs? I wanted to delete this section about control planes, but I figured I should keep it with a note about "If you started using Konnect pre 3.0 gateway, keep the following in mind:" or something like that. What do you think?

cloudjumpercat replied:
Thanks! I'll make those edits

lena-larionova commented on app/konnect/dev-portal/configuration-prep.md:
```diff
@@ -0,0 +1,58 @@
+
+{% mermaid %}
+flowchart TD
+    A{Do you want your users to consume your APIs?} --> B(Yes)
+    B --> C{Are your users internal or external?}
+    C --> |internal| D(Use SSO to allow developers to log in)
+    C --> |external| J(Assign roles to developers after they register)
+    C --> |both| K(Multi-portal with roles for external and SSO for internal)
+    A --> E(No)
+    A --> F(Some)
+    E --> G(Assign a view only role)
+    F --> |they can't consume| G
+    F --> |they can consume| C
+    D --> H{How will you manage developer apps?}
+    J --> H
+    K --> H
+    H --> I(Manually approve or deny apps)
+    H --> L(Automatically approve all apps)
+    I --> M{How will developers authenticate their apps?}
+    M --> N(Dynamic client registration)
+    M --> O(Use default authentication strategy in Konnect)
```
```suggestion
    A{"Do you want your 
    users to consume 
    your APIs?"} --> B("Yes") & E("No") & F("Some")
    B --> C{"Are your users 
    internal or external?"}
    C -- internal --> D("Use SSO to allow 
    developers to 
    log in")
    C -- external --> J("Assign roles 
    to developers 
    after they register")
    C -- both --> K("Multi-portal with roles 
    for external and 
    SSO for internal")
    E --> G("Assign a view only role")
    F -- they can't consume --> G
    F -- they can consume ---> C
    D --> H{"How will 
    you manage 
    developer apps?"}
    J --> H
    K --> H
    H --> I("Manually approve or deny apps") & L("Automatically approve all apps")
    I --> M{"How will developers 
    authenticate 
    their apps?"}
    M --> N("Dynamic client registration") & O("Use default authentication 
    strategy in Konnect")
```
If you add linebreaks, it changes the sizes of the shapes and makes the text bigger/easier to read.

I also reordered the first section a bit to force the arrows into a _slightly_ easier to follow order, though there's a limit to what we can do here.

lena-larionova commented on app/konnect/dev-portal/configuration-prep.md:
```diff
@@ -0,0 +1,58 @@
+    click O "/konnect/dev-portal/applications/enable-app-reg/"
+{% endmermaid %}
+
+## Next steps
+
+Now that you know what your use cases are and which settings you'll need to configure, you can [create your Dev Portal or Dev Portals](/konnect/dev-portal/create-dev-portal). 
```
Is this next steps section necessary? Technically they should be following the steps via links in the diagram.

cloudjumpercat replied:
```suggestion
```
